src/file-upload/dropzone.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './styles.css.js';
+import clsx from 'clsx';
+import InternalIcon from '../icon/internal';
+import InternalBox from '../box/internal';
+
+interface DropzoneProps {
+  onChange: (files: File[]) => void;
+  children: React.ReactNode;
+}
+
+export function useDropzoneVisible() {
+  const [isDropzoneVisible, setDropzoneVisible] = useState(false);
+
+  // Registering global drag events listeners.
+  useEffect(() => {
+    // The timer helps avoiding dropzone blinking.
+    let dragTimer: null | ReturnType<typeof setTimeout> = null;
+
+    // The file-upload dropzone is shown when the user drags files over the to the browser.
+    const onDocumentDragOver = (event: DragEvent) => {
+      event.preventDefault();
+
+      if (event.dataTransfer && event.dataTransfer.types && event.dataTransfer.types.indexOf('Files') !== -1) {
```

Comment:
I think this can be simplified to `if(event.dataTransfer?.types?.includes('Files')) {`

Reply:
Will change to `event.dataTransfer?.types?.indexOf('Files') !== -1`. The `includes` is not possible due to our current transpilation settings.